src/main/java/com/unstoppabledomains/resolution/naming/service/uns/UNS.java

```diff
@@ -0,0 +1,122 @@
+package com.unstoppabledomains.resolution.naming.service.uns;
```

Comment:
@DancingAxolotl Would be great to add tests for new UNS class.

Reply:
The whole class mostly just calls `L2Resolver` and returns the result. There isn't much logic here, so I didn't consider adding tests for it. The `L2Resolver` is covered separately and `Resolution` tests cover the internal UNS logic.

I've added some tests for more complex methods - `batchOwners` and `getTokensOwnedBy`.